src/app/meds-n-labels/activity-2/activity-2.component.ts

```diff
@@ -0,0 +1,161 @@
+
+    /**
+     * Shuffle the array
+     * @param {Array} array the array containing the face images
+     */
+    shuffle(array) {
+        let currentIndex = array.length, temporaryValue, randomIndex;
+        // While there remain elements to shuffle...
+        while (0 !== currentIndex) {
+            // Pick a remaining element...
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            currentIndex -= 1;
+
+            // And swap it with the current element.
+            temporaryValue = array[currentIndex];
+            array[currentIndex] = array[randomIndex];
+            array[randomIndex] = temporaryValue;
+        }
+        return array;
+    }
+
+    getImgElement(card) {
+        return document.getElementById(card).getElementsByTagName('img')[0];
+    }
+
+    /**
+     * Flip card on click
+     * @param {Number} card The index of the clicked element
+     */
+    choose(card) {
+            if (this._clicks === 2) {
+                return;
+            } else {
+                this.getImgElement(card).src = this._baseImgPath + this._faces[card];
+                if (this._clicks === 0) {
+                    this._firstchoice = card;
+                    this._clicks = 1;
+                } else if (this._firstchoice !== card) {
+                    this._clicks = 2;
+                    this._secondchoice = card;
+                    this._obs = Observable.interval(500)
+                            .do(i => this.check());
+                    this._subscription = this._obs.subscribe();
+                }
+            }
+        }
+
+    /**
+     * Check if a match is made
+     */
+    check() {
+        this._subscription.unsubscribe();
+        this._clicks = 0;
+        if (this._faces[this._secondchoice] === this._faces[this._firstchoice]) {
+            const medName = this._faces[this._secondchoice]; // or this.faces[this._firstchoice]
+            const med = medName.substr(0, medName.lastIndexOf('.'));
+            for (let i = 0; i < DIAGNOSIS.length; i++) {
+                if (DIAGNOSIS[i].name === med) {
+                    this._sharedData.customAlert('Congratulations!<br>You matched a pair', DIAGNOSIS[i].desc, 'success');
+                    this.isMatchArr[this._secondchoice] = false;
+                    this.isMatchArr[this._firstchoice] = false;
+                    break;
+                }
+            }
+            this._matches++;
+            if (this._matches === 8) {
+                this._sharedData.customAlert('Good job!', 'You completed this activity!', 'success');
+                this._status = {stage: 3, activity: 2};
+                this._dashboardService.updateProgressStatus(this._status).subscribe(response => {});
+                this.activityComplete = true;
+            }
+        } else {
+            this.getImgElement(this._firstchoice).src = this.backcard;
+            this.getImgElement(this._secondchoice).src = this.backcard;
+            return;
```

Comment:
i'm still trying to understand the need of this `return`, this is the last line of the function, so it automatically returns/terminates. 
also, unnecessary `else`